sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -0,0 +1,484 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.implementation.TablesImpl;
+import com.azure.data.tables.implementation.models.ResponseFormat;
+import com.azure.data.tables.implementation.models.TableEntityQueryResponse;
+import com.azure.data.tables.implementation.models.TablesQueryEntitiesResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.FluxUtil.collectBytesInByteBufferStream;
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+
+/**
+ * class for the table async client
+ */
+@ServiceClient(
+    builder = TableClientBuilder.class,
+    isAsync = true)
+public class TableAsyncClient {
+    private final ClientLogger logger = new ClientLogger(TableAsyncClient.class);
+    private final String tableName;
+    private final TablesImpl impl = null;
+    private final Integer TIMEOUT = 400;
```

Comment:
Should this be a configuration value passed by a customer? I'm guessing if nothing is passed the operation has no timeout. Also is this in seconds, milliseconds, etc?